commit ec38e9d7a4b1f55da6c8e246b7e792cd295d30cd
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2019-05-17T15:58:22+00:00 (5 months ago)
    
    TRITON-1658 32-bit amon-agent should be buildable on x86_64
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/Makefile b/Makefile
index 5255d77..64655ca 100644
--- a/Makefile
+++ b/Makefile
@@ -56,6 +56,22 @@ ENGBLD_REQUIRE		:= $(shell git submodule update --init deps/eng)
 include deps/eng/tools/mk/Makefile.defs
 TOP ?= $(error Unable to access eng.git submodule Makefiles.)
 
+
+#
+# amon currently builds against 32-bit 0.8.x nodejs and depends on old versions
+# of the node-dtrace-provider with a bug that prevents compiling for 32-bit if
+# the *host* is x86_64 (as components do when they include the amon agent).  All
+# fixed versions of node-trace-provider require nodejs >= 0.10.  However,
+# multiple libraries used by amon (zutil, zsock) require nodejs <= 0.8.  This
+# gcc wrapper forces the 32-bit target, effectively inlining the
+# node-trace-provider fix.  This hack should be removed if amon is every updated
+# past nodejs 0.8.  See also TRITON-1658
+#
+# NOTE: This PATH definition must appear before node_prebuilt, which also mucks
+# with PATH.
+#
+export PATH:=$(TOP)/tools/m32-gcc-wrapper:$(PATH)
+
 ifeq ($(shell uname -s),SunOS)
        include deps/eng/tools/mk/Makefile.node_prebuilt.defs
        include deps/eng/tools/mk/Makefile.agent_prebuilt.defs
diff --git a/tools/m32-gcc-wrapper/g++ b/tools/m32-gcc-wrapper/g++
new file mode 100755
index 0000000..46dff0e
--- /dev/null
+++ b/tools/m32-gcc-wrapper/g++
@@ -0,0 +1,12 @@
+#!/bin/sh
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright (c) 2019, Joyent, Inc.
+#
+
+/opt/local/bin/g++ -m32 "$@"
diff --git a/tools/m32-gcc-wrapper/gcc b/tools/m32-gcc-wrapper/gcc
new file mode 100755
index 0000000..f395559
--- /dev/null
+++ b/tools/m32-gcc-wrapper/gcc
@@ -0,0 +1,12 @@
+#!/bin/sh
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+#
+# Copyright (c) 2019, Joyent, Inc.
+#
+
+/opt/local/bin/gcc -m32 "$@"
\ No newline at end of file
